src/wallet/cc_wallet/cc_info.py

```diff
@@ -1,14 +1,22 @@
 from dataclasses import dataclass
 from typing import List, Optional, Tuple
 
+from src.types.program import Program
 from src.types.sized_bytes import bytes32
 from src.util.streamable import streamable, Streamable
+from src.wallet.cc_wallet import cc_wallet_puzzles
 from src.wallet.cc_wallet.ccparent import CCParent
 
 
 @dataclass(frozen=True)
 @streamable
 class CCInfo(Streamable):
-    my_core: Optional[str]  # core is stored as the disassembled string
     parent_info: List[Tuple[bytes32, Optional[CCParent]]]  # {coin.name(): CCParent}
-    my_colour_name: Optional[str]
+    my_colour_name: str
+
+    def puzzle_for_inner_puzzle(self, inner_puzzle: Program) -> Program:
+        genesis_id = bytes.fromhex(self.my_colour_name) if self.my_colour_name else 0
```

Comment:
When does this case happen ?(not self.my_colour_name)

Reply:
Might be left over from when we had `Optional`. I'd love to get rid of it. There still is a case where `my_colour_name` is not being set, and I'd love to get of that too.